director/cache_ads.go

```diff
@@ -35,13 +35,13 @@ import (
 
 type (
 	NamespaceAd struct {
-		RequireToken  bool
-		Path          string
-		Issuer        url.URL
-		MaxScopeDepth uint
-		Strategy      StrategyType
-		BasePath      string
-		VaultServer   string
+		RequireToken  bool         `json:"requireToken"`
+		Path          string       `json:"path"`
+		Issuer        url.URL      `json:"url"`
+		MaxScopeDepth uint         `json:"maxScopeDepth"`
+		Strategy      StrategyType `json:"strategy"`
+		BasePath      string       `json:"basePath"`
+		VaultServer   string       `json:"vaultServer"`
```

Comment:
Does this have any forward/backward compatibility concerns?

Reply:
I don't know. It caused issues with the redirect tests. But it didn't break anything that was already using the NamespaceAds in the code otherwise. As long as it's marshaling properly into a struct, I think it should be okay?

Reply:
I did this and it seems to advertise properly, looking at the debug logs.